src/Controls/src/Core/Handlers/Items/iOS/StructuredItemsViewController.cs

```diff
@@ -86,23 +87,34 @@ protected override CGRect DetermineEmptyViewFrame()
 
 		public override void ViewWillLayoutSubviews()
 		{
-			var hasHeaderOrFooter = _footerViewFormsElement is not null || _headerViewFormsElement is not null;
-			if (hasHeaderOrFooter && CollectionView is MauiCollectionView { NeedsCellLayout: true } collectionView)
+			if (_headerFooterPositionNeedLayoutUpdate)
```

Comment:
This path here is the only interesting part of this PR

The rest of this PR is just reverting a PR that didn't fix the intended issue correctly